web/src/portico/team.ts

```diff
@@ -17,21 +39,24 @@ const repo_name_to_tab_name = {
 
 // Remember the loaded repositories so that HTML is not redundantly edited
 // if a user leaves and then revisits the same tab.
-const loaded_repos = [];
+const loaded_repos: string[] = [];
 
-function calculate_total_commits(contributor) {
+function calculate_total_commits(contributor: Contributor): number {
     let commits = 0;
-    for (const repo_name of Object.keys(repo_name_to_tab_name)) {
-        commits += contributor[repo_name] || 0;
+    for (const repo_name of all_repository_names) {
+        commits += contributor[repo_name];
     }
     return commits;
 }
 
-function get_profile_url(contributor, tab_name) {
+function get_profile_url(contributor: Contributor, tab_name?: string): string | undefined {
     const commit_email_linked_to_github = "github_username" in contributor;
 
     if (commit_email_linked_to_github) {
-        return "https://github.com/" + contributor.github_username;
+        if (contributor.github_username) {
+            return `https://github.com/${contributor.github_username}`;
+        }
+        blueslip.error("Contributor's github username is undefined.");
```

Comment:
This logging is unnecessarily complex; I rewrote this to just check if the field is present in a more sane way.